cylc/flow/scheduler.py

```diff
@@ -1867,12 +1871,11 @@ def check_workflow_timers(self):
         for event, timer in self.timers.items():
             if not timer.timed_out():
                 continue
+            self.run_event_handlers(event)
             abort_conf = f"abort on {event}"
             if self._get_events_conf(abort_conf):
                 # "cylc play" needs to exit with error status here.
                 raise SchedulerError(f'"{abort_conf}" is set')
-            if self._get_events_conf(f"{event} handlers") is not None:
```

Comment:
This was the source of the bug.

In my case I had `mail events = <event>` set, but not `<event> handlers`, this was preventing mail events from running.

I think the workflow events manager already has logic for determining when events should be run so this check was superfluous.